addons/spreadsheet_dashboard/models/spreadsheet_dashboard_group.py

```diff
@@ -9,3 +10,9 @@ class SpreadsheetDashboardGroup(models.Model):
     name = fields.Char(required=True)
     dashboard_ids = fields.One2many('spreadsheet.dashboard', 'dashboard_group_id')
     sequence = fields.Integer()
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_spreadsheet_data(self):
+        for record in self:
+            if record.get_external_id()[record.id] and not record.get_external_id()[record.id].startswith('__export__'):
```

Comment:
- call `.get_external_id` outside the for loop. It doing a `search` internally and it's bad performance-wise in a for loop. You are even calling it twice in the if predicate! So that's 2N `search` instead of a single one
- "record" can be more precise => "group" or "dashboard_group"
```suggestion
    def _unlink_except_spreadsheet_data(self):
        external_ids = self.get_external_id()
        for group in self:
            external_id = external_ids[group.id]
            if external_id and not external_id.startswith('__export__'):
```